src/PathOrderMonotonic.h

```diff
@@ -0,0 +1,449 @@
+            size_t farthest_vertex = getFarthestEndpoint(first, close_line_before->point); //Get to the opposite side.
+            first->start_vertex = farthest_vertex;
+            first->backwards = farthest_vertex != 0;
+            first_endpoint = (*first->converted)[farthest_vertex];
+            lines_before = line_bucket_grid.getNearby(first_endpoint, COINCIDENT_POINT_DISTANCE);
+            close_line_before = std::find_if(lines_before.begin(), lines_before.end(), [first_endpoint, result](SparsePointGridInclusiveImpl::SparsePointGridInclusiveElem<Path*> found_path) {
+                return found_path.val->start_vertex == found_path.val->converted->size() //Don't find any line already in the string.
+                       && vSize2(found_path.point - first_endpoint) < SQUARED_COINCIDENT_POINT_DISTANCE; //And only find close lines.
+            });
+        }
+        while(close_line_after != lines_after.end())
+        {
+            Path* last = close_line_after->val;
+            result.push_back(last);
+            size_t farthest_vertex = getFarthestEndpoint(last, close_line_after->point); //Get to the opposite side.
+            last->start_vertex = (farthest_vertex == 0) ? last->converted->size() - 1 : 0;
+            last->backwards = (farthest_vertex != 0);
+            last_endpoint = (*last->converted)[farthest_vertex];
+            lines_after = line_bucket_grid.getNearby(last_endpoint, COINCIDENT_POINT_DISTANCE);
+            close_line_after = std::find_if(lines_after.begin(), lines_after.end(), [last_endpoint, result](SparsePointGridInclusiveImpl::SparsePointGridInclusiveElem<Path*> found_path) {
+                return found_path.val->start_vertex == found_path.val->converted->size() //Don't find any line already in the string.
+                       && vSize2(found_path.point - last_endpoint) < SQUARED_COINCIDENT_POINT_DISTANCE; //And only find close lines.
+            });
+        }
+
+        //Figure out which of the two endpoints to start with: The one monotonically earliest.
+        const coord_t first_projection = dot(first_endpoint, monotonic_vector);
+        const coord_t last_projection = dot(last_endpoint, monotonic_vector);
+        //If the last endpoint should be printed first (unlikely due to monotonic start, but possible), flip the whole polyline!
+        if(last_projection < first_projection)
+        {
+            std::reverse(result.begin(), result.end());
+            for(Path* path : result) //Also reverse their start_vertex.
+            {
+                path->start_vertex = (path->start_vertex == 0) ? path->converted->size() - 1 : 0;
+                path->backwards = !path->backwards;
+            }
+        }
+
+        if(result.size() == 1)
+        {
+            result[0]->start_vertex = result[0]->converted->size(); //Reset start vertex as "unknown" again if it's not a string of polylines.
+        }
+        return result;
```

Comment:
For consistencies sake: could use a `// Uses RVO!` comment like you did below (or remove that one).

Reply:
I've opted to remove the other one. It's something I'm paying attention to during development, and thus think I need to write down. But for the most part a developer doesn't really need to be reminded of this. Removed here: 57ad843b67160b7055544f98b36cf81416301934

Reply:
Done, so can be resolved :-)